glean-core/src/upload/mod.rs

```diff
@@ -69,6 +77,74 @@ pub fn setup_upload_manager(upload_manager: PingUploadManager) -> Result<()> {
     Ok(())
 }
 
+/// A counter to help limit the number of processing pings per minute.
+///
+/// NOTE: A "processing" ping is a ping that has been dequeued through `get_upload_task`.
+#[derive(Debug)]
+struct PingCounter {
```

Comment:
This doesn't really need to be a different struct, it was easier for me just to isolate the logic I was working on. I can merge it in the `PingUploadManager` if necessary.

Reply:
Yes, for the final thing, I'd rather see this as part of the upload manager, but I have no strong opinion. If it looks cleaner this way, let's stick to it.